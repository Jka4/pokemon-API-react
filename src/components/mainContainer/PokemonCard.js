import React, { lazy, Suspense } from "react";
import Loader from "react-loader-spinner";
import { NavLink } from "react-router-dom";

const ImageContainer = lazy(() => import("@ImageContainer"));

const PokemonCard = (props) => {
  const { src, name, order, base_experience, id, onClick } = props;

  return (
    <NavLink
      to={`/detailedPage/pokemon/${name}`}
      data-testid="testId"
      className="pokemonCard_Outher"
      data-pokemon_id={id}
      onClick={onClick}
      key={id}
    >
      <div className="pokemonCard">
        <div className="imageWrapper">
          {src ? (
            <Suspense
              fallback={
                <Loader type="TailSpin" height={50} width={50} color={"red"} />
              }
            >
              <ImageContainer url={src} cn={"pokemonImage"} />
            </Suspense>
          ) : (
            <Loader type="TailSpin" height={50} width={50} color={"red"} />
          )}
        </div>
        <span>{name}</span>
        <span>{order}</span>
        <span>{base_experience}</span>
      </div>
    </NavLink>
  );
};

export default PokemonCard;
