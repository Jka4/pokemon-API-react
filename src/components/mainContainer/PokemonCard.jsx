import React, { lazy, Suspense } from "react";
import Loader from "react-loader-spinner";
import { NavLink } from "react-router-dom";
import Card from "@material-ui/core/Card";
import PropTypes from "prop-types";

const ImageContainer = lazy(() => import("@ImageContainer"));

const PokemonCard = ({
  src = "",
  name = "",
  order = 0,
  base_experience = 0,
  id = 0,
  onClick = () => {},
}) => {
  return (
    <Card className="pokemonCard_Outher">
      <NavLink
        to={`/detailedPage/pokemon/${name}`}
        data-testid="testId"
        data-pokemon_id={id}
        onClick={onClick}
        key={id}
      >
        <div className="pokemonCard">
          <div className="imageWrapper">
            {src ? (
              <Suspense
                fallback={
                  <Loader
                    type="TailSpin"
                    height={50}
                    width={50}
                    color={"red"}
                  />
                }
              >
                <ImageContainer url={src} cn={"pokemonImage"} />
              </Suspense>
            ) : (
              <Loader type="TailSpin" height={50} width={50} color={"red"} />
            )}
          </div>
          <span>{name}</span>
          <span>{order}</span>
          <span>{base_experience}</span>
        </div>
      </NavLink>
    </Card>
  );
};

PokemonCard.propTypes = {
  base_experience: PropTypes.number,
  id: PropTypes.number,
  name: PropTypes.string,
  onClick: PropTypes.func,
  order: PropTypes.number,
  src: PropTypes.string,
};

export default PokemonCard;
