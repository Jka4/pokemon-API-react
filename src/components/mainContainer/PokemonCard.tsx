import React, { Suspense } from "react";
import { NavLink } from "react-router-dom";
import Card from "@material-ui/core/Card";

type pokemonCardType = {
  base_experience: number;
  id: number;
  name: string;
  order: number;
  src: string;
  placeholderBase64: string;
}

const PokemonCard: React.FC<pokemonCardType> = ({ src, name, order, base_experience, id, placeholderBase64 }: pokemonCardType) => {

  return (
    <Card className="pokemonCard_Outher">
      <NavLink
        to={`/detailedPage/pokemon/${name}`}
        data-pokemon_id={id}
        key={id}
      >
        <div className="pokemonCard">
          <div className="imageWrapper">
            <Suspense fallback={<img loading="lazy" src={placeholderBase64} className="pokemonImage placeholderBase64" alt='' />} >
              <img loading="lazy" src={src} className="pokemonImage" alt="pokemonImage" />
            </Suspense>
          </div>
          <span>{name}</span>
          <span>{order}</span>
          <span>{base_experience}</span>
        </div>
      </NavLink>
    </Card>
  );
};

export default PokemonCard;
