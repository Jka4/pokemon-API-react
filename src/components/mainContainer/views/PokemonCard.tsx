import React from 'react';
import { NavLink } from 'react-router-dom';
import { pokemonCardType } from 'types/index';
import styled from 'styled-components';

import ImageContainer from 'components/ImageContainer/ImageContainer';

const PokemonCard: React.FC<pokemonCardType> = ({ pokeCard }: pokemonCardType) => {
  const { name, order, base_experience, placeholderBase64, image } = pokeCard;

  return (
    <PokeCardWrapper>
      <NavLink to={`/detailedPage/pokemon/${name}`}>
        <PokeCard>
          <Image>
            <ImageContainer
              url={image}
              cn={'deBlur'}
              fallback={<img src={placeholderBase64} className="placeholderBase64 " alt="" />}
            />
          </Image>

          <Span>{name}</Span>
          <Span>{order}</Span>
          <Span>{base_experience}</Span>
        </PokeCard>
      </NavLink>
    </PokeCardWrapper>
  );
};

const PokeCardWrapper = styled.div`
  margin-top: 10px;
  width: 100%;
  height: 100px;
  text-decoration: none;
  color: black;
  overflow: initial !important;
`;

const PokeCard = styled(PokeCardWrapper)`
  border: 2px solid black;
  border-radius: 5px;
  background-color: rgba(250, 235, 215, 0.74);
  cursor: pointer;
  width: 100%;
  height: 100%;
  text-transform: uppercase;
  font-size: 22px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: 1fr;
  grid-column-gap: 20px;
  grid-row-gap: 0px;
  padding-left: 0;
`;

const Image = styled.div`
  height: 20%;
  margin-left: auto;
  margin-right: auto;
  max-width: 100px;
`;

const Span = styled.span`
  text-align: center;
  margin-top: auto;
  margin-bottom: auto;
`;

export default PokemonCard;
