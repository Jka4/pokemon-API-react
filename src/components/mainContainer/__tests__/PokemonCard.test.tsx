import * as React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom';

import PokemonCard from '../views/PokemonCard';

describe('<PokemonCard />', () => {
  let container: any;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it('renders correctly', async () => {
    const pokemon = {
      id: 565,
      image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/565.png',
      imageHQ:
        'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/565.png',
      name: 'carracosta',
      order: 665,
      placeholderBase64:
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABfUlEQVQ4T2NkIBMwYtX3nwEs/p+BgYGREUxhAOwaiXAFVo0BDQoCG+b8+ZmS+014Tvm7JwRtXLWKgTksjOFvzSru0v//2E0Zmb6fbgn73j3zjDFrusnZ38gGoNjYsJ+BpcGR4U/mKqE6FfV3jU9vch/uC9MIYmA4+6ahgYGpoYHhH0wzXGNoaCjz6tWr/zLUBhh3a786oSD5nvH2Kxnm1//Pbu0Pe+cTuiqUeXXY6r8YGhtWabE1hF37VbJfJTlKVGGOwE+er8c+f+N6wb7rg5wlg2gYA8NfhwYHlgMNB/6ANMNsBNH/GRj8pNwbnkx3tXvmx8r6kYFBgIHh5RNFhhXrhVbc4+KpZJiw4wEDKKoYGf5DNDYwMDE0MPyTSbDTffL9v4ez0OtXGpJ/hB9xcqq9fct2//ZHQVYe3v8n73ft3Q1TiwgcqEl4oxBJDWo8gmx+ZszcIMX7/5nkZ0YpNd7/Bw4yMHx+9pnxrJTvX4aGBsxQJSKxoCghO8kBALL4iw/htJjEAAAAAElFTkSuQmCC',
      weight: 810,
    };

    await act(async () => {
      ReactDOM.render(
        <BrowserRouter basename="/">
          <PokemonCard pokeCard={pokemon} />
        </BrowserRouter>,
        await container,
      );
      await expect(container).toMatchSnapshot();
    });
  });
});
