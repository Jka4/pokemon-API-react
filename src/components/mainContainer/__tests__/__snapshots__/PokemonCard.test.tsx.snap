// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonCard /> renders correctly 1`] = `
<div>
  <div
    class="sc-dlfnbm jzyuuj"
  >
    <a
      href="/detailedPage/pokemon/carracosta"
    >
      <div
        class="sc-dlfnbm sc-hKgILt jzyuuj YSRNa"
      >
        <div
          class="sc-eCssSg dOIVFh"
        >
          <img
            alt=""
            class="placeholderBase64 "
            src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAA4AAAAOCAYAAAAfSC3RAAABfUlEQVQ4T2NkIBMwYtX3nwEs/p+BgYGREUxhAOwaiXAFVo0BDQoCG+b8+ZmS+014Tvm7JwRtXLWKgTksjOFvzSru0v//2E0Zmb6fbgn73j3zjDFrusnZ38gGoNjYsJ+BpcGR4U/mKqE6FfV3jU9vch/uC9MIYmA4+6ahgYGpoYHhH0wzXGNoaCjz6tWr/zLUBhh3a786oSD5nvH2Kxnm1//Pbu0Pe+cTuiqUeXXY6r8YGhtWabE1hF37VbJfJTlKVGGOwE+er8c+f+N6wb7rg5wlg2gYA8NfhwYHlgMNB/6ANMNsBNH/GRj8pNwbnkx3tXvmx8r6kYFBgIHh5RNFhhXrhVbc4+KpZJiw4wEDKKoYGf5DNDYwMDE0MPyTSbDTffL9v4ez0OtXGpJ/hB9xcqq9fct2//ZHQVYe3v8n73ft3Q1TiwgcqEl4oxBJDWo8gmx+ZszcIMX7/5nkZ0YpNd7/Bw4yMHx+9pnxrJTvX4aGBsxQJSKxoCghO8kBALL4iw/htJjEAAAAAElFTkSuQmCC"
          />
          <img
            alt="deBlur"
            class="deBlur"
            loading="lazy"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/565.png"
            style="display: none;"
          />
        </div>
        <span
          class="sc-jSgupP fxRBYU"
        >
          carracosta
        </span>
        <span
          class="sc-jSgupP fxRBYU"
        >
          665
        </span>
        <span
          class="sc-jSgupP fxRBYU"
        />
      </div>
    </a>
  </div>
</div>
`;
